src/atomWithMachine.ts

```diff
@@ -30,42 +37,27 @@ export interface MachineAtomOptions<TContext, TEvent extends EventObject> {
   state?: StateConfig<TContext, TEvent>;
 }
 
-type Options<TMachine extends AnyStateMachine> =
-  AreAllImplementationsAssumedToBeProvided<
-    TMachine['__TResolvedTypesMeta']
-  > extends false
-    ? InterpreterOptions &
-        MachineAtomOptions<TMachine['__TContext'], TMachine['__TEvent']> &
-        InternalMachineOptions<
-          TMachine['__TContext'],
-          TMachine['__TEvent'],
-          TMachine['__TResolvedTypesMeta'],
-          true
-        >
-    : InterpreterOptions &
-        MachineAtomOptions<TMachine['__TContext'], TMachine['__TEvent']> &
-        InternalMachineOptions<
-          TMachine['__TContext'],
-          TMachine['__TEvent'],
-          TMachine['__TResolvedTypesMeta']
-        >;
+type Options<TMachine extends AnyStateMachine> = Compute<
+  MachineAtomOptions<ContextFrom<TMachine>, EventFrom<TMachine>> &
+    InternalMachineImplementations<
+      ContextFrom<TMachine>,
+      __ResolvedTypesMetaFrom<TMachine>
```

Comment:
If this `Options` type works, that's great; we're also working on getting rid of typegen types from XState v5 which may simplify this a bit.

Reply:
It seems to be working as far as I can tell too

![image](https://github.com/jotaijs/jotai-xstate/assets/30100875/e1bbcafb-c064-4e83-8009-b03723b8ef74)